cleverhans/dataset.py

```diff
@@ -0,0 +1,146 @@
+"""Dataset class for CleverHans
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from cleverhans.utils_mnist import data_mnist
+
+keras = None  # Only load keras if user tries to use a dataset that requires it
+
+
+class Dataset(object):
+    """Abstract base class representing a dataset.
+    """
+
+    # The number of classes in the dataset. Should be specified by subclasses.
+    nb_classes = None
+
+    def get_factory(self):
+        """Returns a picklable callable that recreates the dataset.
+        """
+
+        if hasattr(self, 'args'):
+            args = self.args
+        else:
+            args = []
+
+        if hasattr(self, 'kwargs'):
+            kwargs = self.kwargs
+        else:
+            kwargs = {}
+
+        return Factory(type(self), args, kwargs)
+
+
+class MNIST(Dataset):
+    """The MNIST dataset"""
+
+    nb_classes = 10
+
+    def __init__(self, train_start=0, train_end=60000, test_start=0,
+                 test_end=10000, center=False):
+        self.kwargs = locals()
```

Comment:
Here you could call  the Dataset constructor
```
Dataset.__init__(self, locals())
```

This way you don't have to copy the `del` statement for every dataset. You can also use the `super` statement if you prefer.

Reply:
Thanks, done